x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/YamlTemplateRegistry.java

```diff
@@ -226,6 +245,14 @@ private IngestPipelineConfig loadIngestPipeline(String name, int version, @Nulla
         );
     }
 
+    // IndexTemplateRegistry ensures that ILM lifecycle policies are not loaded
+    // when in DSL only mode.
+    private LifecyclePolicy loadLifecyclePolicy(String name) {
+        return new YamlLifecyclePolicyConfig(name, "/lifecycle-policies/" + name + ".yaml", this.getClass()).load(
+            LifecyclePolicyConfig.DEFAULT_X_CONTENT_REGISTRY
+        );
+    }
+
```

Comment:
This is the place into which I am suggesting to collapse all of `YamlLifecyclePolicyConfig`. Since that class has no uses outside of here.